metrics-operator/controllers/metrics/controller.go

```diff
@@ -93,7 +93,7 @@ func (r *KeptnMetricReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
 	}
 	// load the provider
-	provider, err2 := providers.NewProvider(metric.Spec.Provider.Name, r.Log, r.Client)
+	provider, err2 := providers.NewProvider(metric.Spec.Provider.Name, metricProvider.Spec.Type, r.Log, r.Client)
```

Comment:
Optional in my opinion, but to make this a little bit cleaner, we could add a `GetType` method to the `KeptnMetricsProvider` type, and pass the value of that to the `NewProvider` method, instead of the name and the type - this way we would not have to do the check for the name/type within the `NewProvider` function.

Reply:
nice suggestion, i implemented it :)